modelforge/potential/utils.py

```diff
@@ -359,18 +242,22 @@ class FeaturizeInput(nn.Module):
         "spin_state",
     ]
 
-    def __init__(self, featurization_config: Dict[str, Union[List[str], int]]) -> None:
+    def __init__(self, featurization_config: Dict[str, Dict[str, int]]) -> None:
         """
         Initialize the FeaturizeInput class.
 
-        For per-atom non-categorical properties and per-molecule properties (both categorical and non-categorical), we append the embedded nuclear charges and mix them using a linear layer.
+        For per-atom non-categorical properties and per-molecule properties
+        (both categorical and non-categorical), we append the embedded nuclear
+        charges and mix them using a linear layer.
 
-        For per-atom categorical properties, we define an additional embedding and add the embedding to the nuclear charge embedding.
+        For per-atom categorical properties, we define an additional embedding
+        and add the embedding to the nuclear charge embedding.
 
         Parameters
         ----------
         featurization_config : dict
-            A dictionary containing the featurization configuration. It should have the following keys:
+            A dictionary containing the featurization configuration. It should
+            have the following keys:
             - "properties_to_featurize" : list
```

Comment:
The structure of the toml seems to drop "properties_to_featurize".  I think we need to have this so that we know whether to throw an error if the properties about those features are not set.

Reply:
You are absolutely right; this has been addressed!